README.md

```diff
@@ -1,2 +1,86 @@
+
+## Full Project Specification
+
+### Backend stack
+
+The server application needs to be written in `Python 3` using `Django 1.9`. Data should be retrieved asynchronously using `Celery` on an interval scheduled with `Celerybeat`. For this project, Django can serve as both the Celery Broker and result backend.
+
+Data retrieved from Twitter should be stored using the Django ORM in a `PostgreSQL` database, and should be served up to the UI via a RESTful API built with `Django Rest Framework`.
+
+### UI Stack
+
+The UI should be a modern and standards compliant web application using `Bootstrap 3` as a CSS framework. You have the option to choose whichever Javascript framework you prefer to build the application. You will get bonus points for using `React.js` or `EmberJS`
+
+Data visualizations can be implemented using `Flot`, `d3.js` (including derivatives like `NVD3`), or `Chart.js`. Front-end build systems are entirely optional, but we’ve been enjoying `webpack` lately.
+
+### Functionality
+
+The UI application should require basic authentication (a username and password) and have the ability to create new user accounts on demand. Django’s built-in authentication system is fine, so long as it is configured to work with `Django Rest Framework` correctly.
+
+Twitter data should be stored in a single model/database table. The schema is up to you, but aggregations should happen at runtime either at the database level or in Python if you can optimize the performance well enough.
+
+The application should consist of three views. These views can be implemented as an SPA (Single Page Application) in JavaScript or as three separate `Django` views. There is no penalty or benefit to choosing one or the other.
+
+* Login view
+
+  Unauthenticated users are directed here
+
+* Dashboard view
+
+  Authenticated users are directed here and can view the dashboard
+
+* User management view
+
+  User account CRUD happens here.
+
+All data interaction with the backend should occur over the REST API
+
+### Workflow & Infrastructure
+
+To get started on `The Project`, fork this respository on GitHub and do all work inside that forked respository.
+
+The application (`Django` and `Celery`) should be deployed to an `Amazon EC2` server. A `t2.micro` instance will be more than adequate. The instance should run Ubuntu 14.04 and the database should be deployed to Amazon RDS. You should be able to create all of these instances on Amazon Web Service's free tier at no cost to you.
+
+`Django` should be deployed using `uWSGI`, which should be running behind `Nginx` via a reverse proxy.
+
+### Get Started!
+
+Good luck, and be sure to check the resources section below for further reading materials!
+
+### Resources
+
+In addition to the Lofty Labs Slack channel, here are some good places to start when researching the various tools needed to build this application.
+
+* [Very good beginner's Python reference](http://learnpythonthehardway.org/book/)
```

Comment:
@tycoles any resources you'd like to add that you've found are good for beginners?